src/PoolManager.sol

```diff
@@ -185,11 +194,13 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
             })
         );
 
-        _accountPoolBalanceDelta(key, delta);
-
         emit ModifyLiquidity(id, msg.sender, params.tickLower, params.tickUpper, params.liquidityDelta);
 
-        key.hooks.afterModifyLiquidity(key, params, delta, hookData);
+        BalanceDelta hookDelta = key.hooks.afterModifyLiquidity(key, params, delta, hookData);
+        delta = delta - hookDelta;
```

Comment:
Hm why is this only allowed on after again?

Reply:
because in the `before` call you dont know how many units of currency0 and currency1 are actually being sent/removed from the pool. So theres kind of nothing to edit at that point - only in the after call do you know what they are